MEMMODEL_CONSTRAINTS.md

```diff
@@ -103,6 +103,19 @@ may perceive in that statement.)
   for memory used for atomics.  The situation is somewhat better in
   well-typed C++ but reality is complicated.
 
+## Implementation targeting C and C++ or LLVM
+
+* _Targeting C or C++_: it should be possible to implement
+  SharedArrayBuffer using C or C++.  The C/C++11 memory model allows
+  accesses with a number of orderings, and there should be a mapping
+  from SharedArrayBuffer accesses to C/C++11 orderings. Racy
+  SharedArrayBuffer accesses are required to produce defined values,
+  so cannot be implemented using C/C++ non-atomics.
+* _Targeting LLVM_: implementations may also decide to target
+  LLVM, which may provide for more optimization than targeting C/C++
+  and compiling to LLVM. In particular, LLVM allows unordered access,
+  which does not require per-variable sequential consistency.
```

Comment:
>SAB only uses SC, but GetValueFromBuffer can call ReadSharedMemory with accessAtomic set to false, and so result in non-atomic access. I don't think there's a coherence requirement for these non-atomics, is there?

Correct. To be more precise: we likely want to enable most optimizations which JS engines already do or could do when accessing SAB non-atomically, except where that would cause races in a SAB where no race existed before (this can happen e.g. when the optimizer speculates).